pilot/pkg/features/pilot.go

```diff
@@ -24,6 +24,16 @@ import (
 	"istio.io/pkg/env"
 )
 
+const (
+	// DefaultInboundCiphers for server side TLS configuration.
+	DefaultInboundCiphers string = "ECDHE-ECDSA-AES256-GCM-SHA384," +
```

Comment:
Seems like this list is the same as the originally configured set, moved from below

Can we expand the comment to describe why this list is chosen by default and if necessary why the list is in a certain order

If this list would meet specific compliance requirements it might be useful to capture here. The advantage of capturing in MeshConfig is we can use generated docs to convey this important information to users.

Reply:
Is this list used only between sidecars or also for the default gateway configuration?

In any case, we should use a better source, e.g. "RESTRICTED" profile from:
https://cloud.google.com/load-balancing/docs/ssl-policies-concepts

Note that AES128 should be preferred over AES256, since both are perfectly safe, and there is no need to use the more CPU-intensive crypto here.

Reply:
Thanks! Updated the default cipher suites based on the "RESTRICTED" profile from
https://cloud.google.com/load-balancing/docs/ssl-policies-concepts
and the Envoy BoringSSL FIPS cipher list on
https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/transport_sockets/tls/v3/common.proto.html.

Reply:
If we are going ahead with this PR of changing defaults we should update the doc and release notes it https://istio.io/latest/docs/concepts/security/#mutual-tls-authentication. We should control the default change via flag so that it does not fail with upgrades.